deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -1781,3 +1799,16 @@ extension ChatViewController: ChatInputTextViewPasteDelegate {
         sendSticker(image)
     }
 }
+
+// MARK: - UINavigationControllerDelegate
+extension ChatViewController: UINavigationControllerDelegate {
+    override func willMove(toParent parent: UIViewController?) {
+        if parent == nil {
+            //logger.debug("chat observer: remove")
+            removeObservers()
+        } else {
+            //logger.debug("chat observer: setup")
+            setupObservers()
```

Comment:
so, the trick here is that the observer stays active all time time the chat is present?

but why not just do that in init()/deinit() then?

in case not every willMove(parent) is paired with a willMove(nil) we will have the too-many-observers issue again.

Reply:
because deinit is not necessarily called immediately. This can lead to a situation when multiple observers are still registered, so that methods are called twice

Reply:
> so, the trick here is that the observer stays active all time time the chat is present?

present as in present in the navigation stack. willMove is always called if the VC is added or removed from the navigation stack, no matter if the VC is visible to the user or somewhere hidden because other VCs are on top of it.

Reply:
> in case not every willMove(parent) is paired with a willMove(nil) we will have the too-many-observers issue again.

from what I understood willMove is always paired